tools/individual-packages-gen/gen-phase1.ts

```diff
@@ -0,0 +1,209 @@
+          const alphaDevDependencies: { [dep: string]: string } = {
+            // we need jsii-rosetta in the dependencies,
+            // otherwise the binary cannot be found when running the 'extract' script
+            'jsii-rosetta': jsiiRosettaVersion,
+          };
+          const devDependencies = srcPackageJson.devDependencies || {};
+          for (const devDependency of Object.keys(devDependencies)) {
+            if (devDependency.startsWith('@aws-cdk/')) {
+              delete devDependencies[devDependency];
+            }
+            if (alphaPackages[devDependency]) {
+              alphaDevDependencies[alphaPackages[devDependency]] = pkg.version;
+            }
+          }
+          devDependencies['@aws-cdk/assert'] = pkg.version;
+          devDependencies['aws-cdk-lib'] = pkg.version;
+          devDependencies.constructs = '^10.0.0';
+          // we save the devDependencies in a temporary key in package.json
+          srcPackageJson.tmp_devDependencies = {
+            ...devDependencies,
+            ...alphaDevDependencies,
+            ...alphaDependencies,
+          };
+          srcPackageJson.devDependencies = {
+            ...alphaDevDependencies,
+            ...alphaDependencies,
+          };
+
+          // peer dependencies
+          srcPackageJson.tmp_peerDependencies = {
+            'aws-cdk-lib': `^${pkg.version}`,
+            'constructs': '^10.0.0',
+            ...(Object.keys(alphaDependencies)
+              .reduce((acc, alphaDependency) => {
+                acc[alphaDependency] = `^${pkg.version}`;
+                return acc;
+              }, {} as { [dep: string]: string })),
+          };
+          srcPackageJson.peerDependencies = undefined;
+
+          // turn off the L1 generation, which uses @aws-cdk/ modules
+          delete srcPackageJson.scripts.gen;
+
+          fileProcessed = true;
+          fs.writeJsonSync(destination, srcPackageJson, { spaces: 2 });
+        }
+      } else if (sourceFileName === '.gitignore') {
+        // ignore everything, otherwise there are uncommitted files present in testing,
+        // because the module's .gitignore file has entries like !.eslintrc.js
+        fileProcessed = true;
+        const gitIgnoreContents = fs.readFileSync(source);
+        fs.outputFileSync(destination, Buffer.concat([gitIgnoreContents, Buffer.from('\n*\n')]));
+      } else if (sourceFileName === '.eslintrc.js') {
+        // Our ESLint configuration prevents importing libraries not in 'dependencies',
+        // which is against our philosophy for experimental modules (which use peerDependencies).
+        // Given that, remove that rule from the ESLint configuration of that module
+        fileProcessed = true;
+        const esLintRcContents = fs.readFileSync(source);
+        fs.outputFileSync(destination, Buffer.concat([esLintRcContents,
+          Buffer.from("\ndelete baseConfig.rules['import/no-extraneous-dependencies'];\n")]));
```

Comment:
But they should be declared as `devDependencies` which should satisfy the eslint rule, no?

Reply:
The ESLint rule only considers direct dependencies, not `devDepenencies`.

Reply:
The docs state otherwise.

https://github.com/import-js/eslint-plugin-import/blob/master/docs/rules/no-extraneous-dependencies.md

> Forbid the import of external modules that are not declared in the package.json's dependencies, devDependencies, optionalDependencies, peerDependencies, or bundledDependencies

Reply:
Changed the configuration to `['error', { devDependencies: false, peerDependencies: false } ]` instead of turning it off.